htlcswitch/hop/blinding.go

```diff
@@ -85,7 +85,8 @@ func EncodeBlindedRouteData(data *BlindedRouteData) ([]byte, error) {
 }
 
 // DecodeBlindedRouteData decodes the data provided within a blinded route.
-func DecodeBlindedRouteData(r io.Reader) (*BlindedRouteData, error) {
+func DecodeBlindedRouteData(r io.Reader,
```

Comment:
I'm not a big fan of the pattern we currently have of coupling decoding + validation - it's messy and I don't know why we do it. In my other life as an anti-jamming crusader I've though about [refactoring this out](https://github.com/lightningnetwork/lnd/pull/7339/commits/812cec9a00cfebfac42eca4cddc6757c97dc72eb). 

I think the we should break the bad pattern and just return the `tlvMap` along with this function so that we can do decoding separately.

Reply:
Okay! Question for you: Do you think one could determine validity based on just the returned structure, without having to have access to the raw parsed TLV type map? After all, the types we parse should be reflected in the contents of the returned struct. I suppose most generally, not all types in the TLV need to be included in the structure, so maybe it would be safer to just use the raw parsed type map.

Reply:
Ehh nvm. The handling of 0 values might make that more trouble than it's worth. The struct having a 0 value doesn't inform whether the TLV included that 0 value or not. Unless you want to use pointers for every struct field and adopt the convention `nil = not parsed',  it's probably easier to just return the tlv type map.

Reply:
Yeah zero values will kill you every time :(